src/protocols/secure_channel/CASESession.h

```diff
@@ -213,6 +305,7 @@ class DLL_EXPORT CASESession : public Messaging::UnsolicitedMessageHandler,
 
 private:
     friend class TestCASESession;
+    friend class Testing::FuzzCASESession;
```

Comment:
You should not need to friend. If the new APIs are public APIs, or protected, they will be fine to use.

Protected would require you to do basic subclassing of CASESession in the test.

Public is a bit easier, but uglier.

Suggest you make the new methods for generation/parsing at least protected and add wrapper methods in the test class that derives from CASESession to just call these protected members. That avoids needing to friend things which is a bad pattern that basically makes everything public in the test which is worse.

Reply:
Yes I agree that using `friend` is a bad patter.
I could remove it for the `FuzzCASESession`. However, the `TestCASESession` class used in Unit Tests, is using more than just the generation/parsing of Sigma1/Sigma2, thus making it non-friend would require other changes in CASESession class (such as making some function private functions protected, etc). (I could do those changes too, I dont know if they should be in this same PR though)

Reply:
- I made the new methods for generation/parsing (and the existing ParseSigma1) protected.
- I added a Wrapper class in TestCASESession to test them.